modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -1587,12 +1587,14 @@ bool CvVideoWriter_GStreamer::writeFrame( const IplImage * image )
 
     handleMessage(pipeline);
 
+#if GST_VERSION_MAJOR > 0
     if (input_pix_fmt == GST_VIDEO_FORMAT_ENCODED) {
         if (image->nChannels != 1 || image->depth != IPL_DEPTH_8U || image->height != 1) {
             CV_ERROR(CV_StsUnsupportedFormat, "cvWriteFrame() needs images with depth = IPL_DEPTH_8U, nChannels = 1 and height = 1.");
         }
     }
-    else if(input_pix_fmt == GST_VIDEO_FORMAT_BGR) {
```

Comment:
No, just else without `{}`. There should be "else if" statement as a result (without preprocessor's `#if` / `#endif`)